### DISCUSSION TITLE
Ask for help on how to express complex expression

### DISCUSSION THREAD

**Discussion by deerball** [optimatika/ojAlgo#524] [General, ojAlgo and related code]
Assume that there are 2 variables x, y.  

I'd like to know how to use org.ojalgo.optimisation.Expression to express [(2 * x + 2 * 3) - 5] + [(3 * y + 3 * 4) - 10] < 100.

Or, is there any other way to express the expression above?

**Comment by apete** [comment]
Not directly. You have to expand/simplify the expression, and adjust the upper limit for the constant part.

All you can do is `set(variable, value)` or `add(variable, value)`, and similar for quadratic terms.

**Reply by deerball** [reply]
Thanks, got it!

for expression [(2 * x + 2 * 3) - 5] + [(3 * y + 3 * 4) - 10] < 100, I transform it into 
=> (2 * x + 2 * 3) + (3 * y + 3 * 4) < 100 + 5 + 10
=> (2 * x) + (3 * y) < 115 - 2 * 3 - 3 * 4
=> (2 * x) + (3 * y) < 97

then I can use set(x, 2).set(y, 3).upper(97), right?

**Reply by deerball** [reply]
Also, in your reply you mentioned method add(variable, value). 

What does it use for? 

Use set(x, 2).add(x, 6) To express 2 * x + 6?
